make/RunTests.gmk

```diff
@@ -36,6 +36,24 @@ include FindTests.gmk
 # Parse global control variables
 ################################################################################
 
+HASH:=\#
+
+define IfAppend
+$(if $(strip $1),$(strip $1)$(strip $2),)
+endef
+
+define IfPrepend
+$(if $(strip $1),$(strip $2)$(strip $1),)
+endef
+
+define TestID
+$(subst .java,,$(subst .java$(HASH),,$(suffix $(notdir $1))))
+endef
```

Comment:
This macro deserves a comment explaining what it's trying to do.

Reply:
I added a description and made it a one-liner as well